src/transformers/trainer.py

```diff
@@ -1622,6 +1617,24 @@ def _inner_training_loop(
         self.state = TrainerState()
         self.state.is_hyper_param_search = trial is not None
 
+        # Compute absolute values for logging, eval, and save if given as ratio
+        # Also stores the ratios in `self._step_ratio` on the first pass
+        if args.logging_steps is not None:
+            if args.logging_steps < 1:
+                self._step_ratios["logging_steps"] = float(args.logging_steps)
```

Comment:
Why do we need to case to float here if it's already < 1 ?

Reply:
Good catch!